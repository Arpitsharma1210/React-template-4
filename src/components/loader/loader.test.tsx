import React from "react";
import { render, screen } from "@testing-library/react";
import Loader from "./index";
import { useSelector } from "react-redux";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

describe("Loader component", () => {
  beforeEach(() => {
    (useSelector as jest.Mock).mockClear();
  });

  it("renders loader when visibility is true", () => {
    (useSelector as jest.Mock).mockReturnValue({ visibility: true });

    render(<Loader />);

    expect(screen.getByAltText("noah logo loader")).toBeInTheDocument();
  });

  it("does not render loader when visibility is false", () => {
    (useSelector as jest.Mock).mockReturnValue({ visibility: false });

    render(<Loader />);

    expect(screen.queryByAltText("noah logo loader")).not.toBeInTheDocument();
  });

  it("matches snapshot when loader is visible", () => {
    (useSelector as jest.Mock).mockReturnValue({ visibility: true });

    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when loader is not visible", () => {
    (useSelector as jest.Mock).mockReturnValue({ visibility: false });

    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
