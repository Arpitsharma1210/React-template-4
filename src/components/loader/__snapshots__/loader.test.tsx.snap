// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches snapshot when loader is not visible 1`] = `<DocumentFragment />`;

exports[`Loader component matches snapshot when loader is visible 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh ispqzj"
  >
    <picture>
      <img
        alt="noah logo loader"
        src="test-file-stub"
      />
    </picture>
  </div>
</DocumentFragment>
`;
